### PR TITLE
feat: block pending meetings for requires confirmation

### PR DESCRIPTION
**Pull request by sean-brydon**
## What does this PR do?
This PR gives the user the choice to enable "requires confirmation" events to block the calendar when they are pending
CleanShot 2024-08-23 at 14 04 40

## Mandatory Tasks (DO NOT REMOVE)

- [X] I have self-reviewed the code (A decent size PR without self-review might be rejected).
- [ ] I have added a Docs issue [here](https://github.com/calcom/docs/issues/new) if this PR makes changes that would require a [documentation change](https://docs.cal.com/docs). If N/A, write N/A here and check the checkbox.
- [X] I confirm automated tests are in place that prove my fix is effective or that my feature works.

## How should this be tested?
1) Create an event type with requires confirmation
2) Enable the new checkbox
3) Place a test booking with this event type
4) Notice even tho the event is pending - slot gets blocked
5) Meanwhile create a new event type that has requires confirmation but dont check this box
6) Make a test booking
7) Ensure that that slot is still bookable (as the block slots setting is disabled) 
8) profit

### CODE REVIEW

File: apps/web/components/eventtype/RequiresConfirmationController.tsx [hunk 1]
```diff
@@ -121,73 +128,86 @@ export default function RequiresConfirmationController({
...
-                                      )}>
-                                      <Select
-                                        inputId="notice"
-                                        options={options}
-                                        isSearchable={false}
-                                        isDisabled={requiresConfirmationLockedProps.disabled}
-                                        innerClassNames={{ control: "rounded-l-none bg-subtle" }}
-                                        onChange={(opt) => {
+                      <>
+                        <RadioField
```
Comment (sean-brydon) [inline apps/web/components/eventtype/RequiresConfirmationController.tsx:132]: This was moved into a fragment so theres a big diff here. All that was changed was there was a checkbox added to the bottom of the radioField

File: packages/core/getBusyTimes.ts [hunk 2]
```diff
@@ -124,6 +125,18 @@ export async function getBusyTimes(params: {
...
+              endTime: { gte: startTimeDate },
+              eventType: {
+                id: eventTypeId,
+                requiresConfirmation: true,
+                requiresConfirmationWillBlockSlot: true,
+              },
+              status: {
+                in: [BookingStatus.PENDING],
+              },
+            },
```
Comment (sean-brydon) [inline packages/core/getBusyTimes.ts:139]: IF the event type ID is the event type we are booking from AND requires confirmation AND block slot. Pull back pending bookings between the time range

File: packages/core/getBusyTimes.ts [hunk 3]
```diff
@@ -124,6 +125,18 @@ export async function getBusyTimes(params: {
                 },
               },
             },
+            {
+              startTime: { lte: endTimeDate },
+              endTime: { gte: startTimeDate },
+              eventType: {
+                id: eventTypeId,
+                requiresConfirmation: true,
```
Comment (emrysal) [inline packages/core/getBusyTimes.ts:133]: I'm wondering if we should do this in a filter; it depends on how much of a performance hit this is

**Review by emrysal — DISMISSED** [review]
We may consider doing this in filter fashion instead of in the query - but for now this is fine. 🚀

**Comment by CarinaWolli** [comment]
@sean-brydon handleChildrenEventTypes.test.ts unit test is failing

**Comment by sean-brydon** [comment]
> @sean-brydon handleChildrenEventTypes.test.ts unit test is failing

Makes sense! Forgot to include it in the expect statement. Fixing

**Review by Udit-takkar — COMMENTED** [review]
LGTM. 

Blocking ?:- https://github.com/calcom/cal.com/pull/16335/files#r1729133856

**Review by CarinaWolli — APPROVED** [review]

**Comment by CarinaWolli** [comment]
> LGTM.
> 
> Blocking ?:- [#16335 (files)](https://github.com/calcom/cal.com/pull/16335/files#r1729133856)

Not blocking, @emrysal approved with this comment: https://github.com/calcom/cal.com/pull/16335#pullrequestreview-2257496605

**Pull request merged**